build/azDevops/azure/templates/steps/build-java.yml

```diff
@@ -146,15 +167,56 @@ steps:
 
   # SonarCloud
   - ${{ if eq(parameters.static_code_analysis, true) }}:
+      - bash: |
+          set -euxo pipefail
+          function strip_refs()
+          {
+            local BRANCH_REF="$1"
+
+            RETURN_BRANCH="$(sed -e "s%^refs/\(heads\|tags\)/%%" <<< $BRANCH_REF)"
```

Comment:
why don't you just echo this out as opposed to making a global variable?

Reply:
Could do, but echoing is prone to failures, if the method did anything else. Plus we have `-x` set which means we'd get the command running output in the output of the function iirc.